README.md

```diff
@@ -1,5 +1,149 @@
+fix this.
+
+This process takes longer than transfers the other way around because they get
+relayed in batches rather than individually. It primarily depends on the amount
+of transfers of the same token and the fees the senders are paying.
+
+Important notice: if an "unlisted" (with no monetary value) ERC20 token gets
+sent into Umee it won't be possible to transfer it back to Ethereum, unless a
+validator is configured to batch and relay transactions of this token.
+
+### Send a transfer from Ethereum to Umee
+
+Any ERC20 token can be sent to Umee and it's done using the command
+`peggo bridge send-to-cosmos`, use the `--help` flag for more information. It
+can also be done by calling the `sendToCosmos` method on the Peggy contract.
+
+The ERC20 tokens will be locked in the Peggy contract and new coins will be
+minted on Umee with the denomination `peggy{token_address}`. This process takes
+around 3 minutes or 12 Ethereum blocks.
+
+## How it works
+
+Peggo allows transfers of assets back and forth between Ethereum and Umee.
+It supports both assets originating on Umee and assets originating on Ethereum
+(any ERC20 token).
+
+It works by scanning the events of the contract deployed on Ethereum (Peggy) and
+relaying them as messages to the Umee chain; and relaying transaction batches and
+validator sets from Umee to Ethereum.
+
+### Events and messages observed/relayed
+
+#### Ethereum
+
+**Deposits** (`SendToCosmosEvent`): emitted when sending tokens from Ethereum to
+Umee using the `sendToCosmos` function on Peggy.
+
+**Withdraw** (`TransactionBatchExecutedEvent`): emitted when a batch of
+transactions is sent from Umee to Ethereum using the `submitBatch` function on
+the Peggy contract by a validator. This serves as a confirmation to Umee that
+the batch was sent successfully.
+
+**Valset update** (`ValsetUpdatedEvent`): emitted on init of the Peggy contract
+and on every execution of the `updateValset` function.
+
+**Deployed ERC 20** (`ERC20DeployedEvent`): emitted when executing the function
+`deployERC20`. This event signals Umee that there's a new ERC20 deployed from
+Peggy, so Umee can map the token contract address to the corresponding native
+coin. This enables transfers from Umee to Ethereum.
+
+#### Umee
+
+ **Validator sets**: Umee informs the Peggy contract who are the current
+ validators and their power. This results in an execution of the `updateValset`
+ function.
+
+ **Request batch**: Peggo will check for new transactions in the Outgoing TX Pool
+ and if the transactions' fees are greater than the set minimum batch fee, it
+ will send a message to Umee requesting a new batch.
+
+ **Batches**: Peggo queries Umee for any batches ready to be relayed and relays
+ them over to Ethereum using the `submitBatch` function on the Peggy contract.
+
+### Transfers from Umee to Ethereum
+
+WIP (describe in simple words how a transfer from Umee to Ethereum works)
```

Comment:
Don't we describe this above?

Reply:
I wanted to get a bit more technical here, like describe in more details how the loops work and that stuff

Reply:
I'll remove it for now so it looks cleaner, and then I'll circle back to this